streams-tests/shared/src/test/scala/zio/stream/ZStreamSpec.scala

```diff
@@ -2415,6 +2415,34 @@ object ZStreamSpec extends ZIOBaseSpec {
             _     <- TestClock.setTime(210.milliseconds)
             value <- fiber.join
           } yield assert(value)(equalTo(List(0 -> 0, 0 -> 1, 1 -> 1, 1 -> 2)))
+        },
+        testM("zipWithNext") {
+          for {
+            result0 <- ZStream(1, 2, 3).zipWithNext.runCollect
```

Comment:
Each of 3 functions have a property that determines them
e.g. running `zipWithNext` is equivalent to running the original stream and performing `list.zip(list.tail)` afterwards.

Can it be turned into property-based tests?

Reply:
@simpadjo true that, will update these.

Reply:
@simpadjo done